compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/input/pointer/PointerIcon.kt

```diff
@@ -73,21 +72,29 @@ fun Modifier.pointerHoverIcon(icon: PointerIcon, overrideDescendants: Boolean =
         if (pointerIconService == null) {
             Modifier
         } else {
-            this.pointerInput(icon, overrideDescendants) {
+            val rememberIcon = rememberUpdatedState(icon)
+            val rememberOverrideDescendants = rememberUpdatedState(overrideDescendants)
+
+            // Create and chain a new instance of a dumbModifier for every new icon.
+            // This forces the LayoutNode to requestRelayout, leading to a synthetic event being dispatched
+            // and then handled here in pointerInput below, and then applying the icon to the pointerIconService.
+            val dumbModifier = remember(icon) { object : Modifier.Element {} }
```

Comment:
This uses implementation details of the how relayout happens. In the future there can be an optimisation to not perform relayout on new empty modifiers.

What if we tell explicitly, that we need a new synthetic event?
```
SideEffect {
   pointerIconService.refresh()
}

override val pointerIconService = object : PointerIconService {
    override var current: PointerIcon = PointerIconDefaults.Default

    fun refresh() = pointerPositionUpdater.onLayout() // better to rename onLayout to `needUpdate`, because now we call it not only on layout
}
```

Also, if it is not complicated, better to write a test to this behaviour.

Reply:
> We also can add DisposableEffect conditionally: only when a component is hovered: if (isHovered) DisapobleEffect.

Great idea ). Let's do this way